src/vtab/csvtab.rs

```diff
@@ -58,6 +59,7 @@ struct CsvTab {
     base: ffi::sqlite3_vtab,
     /// Name of the CSV file
     filename: String,
+    data: String,
```

Comment:
I am not sure but `filename` and `data` are exclusive.
So maybe we should introduce something like:
```rust
enum Source {
  File(String),
  Data(String),
}

impl Read for Source {
...
}
```